pkg/runtime/runtime.go

```diff
@@ -1878,6 +1913,272 @@ func (a *DaprRuntime) initNameResolution() error {
 	return nil
 }
 
+func (a *DaprRuntime) sendBulkToDeadLetter(
+	name string, msg *pubsub.BulkMessage, deadLetterTopic string,
+	bulkResponses []pubsub.BulkSubscribeResponseEntry,
+) error {
+	data := make([]pubsub.BulkMessageEntry, 0)
```

Comment:
You can pre-allocate this since you know the size for avoiding lots of allocations in a hot path.

```
data := make([]pubsub.BulkMessageEntry, len(msg.Entries))
```

then:

```
n := 0
for _, message := range msg.Entries {
    ...
    if bulkResponses == nil || bulkResponses[entryID].Error != nil {
		data[n] = message
    }
}
data = data[:n]
```

Reply:
Done